modules/33acrossAnalyticsAdapter.js

```diff
@@ -0,0 +1,461 @@
+import buildAdapter from '../libraries/analyticsAdapter/AnalyticsAdapter.js';
+import adapterManager from '../src/adapterManager.js';
+import CONSTANTS from '../src/constants.json';
+
+/**
+ * @typedef {typeof import('../src/constants.json').EVENTS} EVENTS
+ */
+const { EVENTS } = CONSTANTS;
+
+const ANALYTICS_VERSION = '1.0.0';
+const PROVIDER_NAME = '33across';
+const DEFAULT_TRANSACTION_TIMEOUT = 3000;
+const DEFAULT_ENDPOINT = `${window.origin}/api`; // TODO: Update to production endpoint
+
+export const log = getLogger();
+
+/**
+ * @typedef {Object} AnalyticsReport - Sent when all bids are complete (as determined by `bidWon` event)
+ * @property {string} analyticsVersion - Version of the Prebid.js 33Across Analytics Adapter
+ * @property {string} pid - Partner ID
+ * @property {'pbjs'} src - Source of the report
+ * @property {string} pbjsVersion - Version of Prebid.js
+ * @property {Auction[]} auctions
+ */
+
+/**
+ * @typedef {Object} AnalyticsCache
+ * @property {string} pid Partner ID
+ * @property {Object<string, Auction>} auctions
+ * @property {Object<string, Bid[]>} bidsWon
+ */
+
+/**
+ * @typedef {Object} Auction - Parsed auction data
+ * @property {AdUnit[]} adUnits
+ * @property {string} auctionId
+ * @property {string[]} userIds
+ */
+
+/**
+ * @typedef {`${number}x${number}`} AdUnitSize
+ */
+
+/**
+ * @typedef {('banner'|'native'|'video')} AdUnitMediaType
+ */
+
+/**
+ * @typedef {Object} BidResponse
+ * @property {number} cpm
+ * @property {string} cur
+ * @property {number} [cpmOrig]
+ * @property {number} cpmFloor
+ * @property {AdUnitMediaType} mediaType
+ * @property {AdUnitSize} size
+ */
+
+/**
+ * @typedef {Object} Bid - Parsed bid data
+ * @property {string} bidder
+ * @property {string} bidId
+ * @property {string} source
+ * @property {string} status
+ * @property {BidResponse} [bidResponse]
+ * @property {1|0} [hasWon]
+ */
+
+/**
+ * @typedef {Object} AdUnit - Parsed adUnit data
+ * @property {string} transactionId - Primary key for *this* auction/adUnit combination
+ * @property {string} adUnitCode
+ * @property {string} slotId - Equivalent to GPID. (Note that
+ * GPID supports adUnits where multiple units have the same `code` values
+ * by appending a `#UNIQUIFIER`. The value of the UNIQUIFIER is likely to be the div-id,
+ * but, if div-id is randomized / unavailable, may be something else like the media size)
+ * @property {Array<AdUnitMediaType>} mediaTypes
+ * @property {Array<AdUnitSize>} sizes
+ * @property {Array<Bid>} bids
+ */
+
+/**
+ * After the first bid is initiated, we wait until every bid is completed before sending the report.
+ *
+ * We will listen for the `bidWon` event to determine when all bids are complete.
+ */
+class TransactionManager {
+  #timeoutId;
+  #pending = 0;
+  #timeout;
+  #transactions = {};
+  #onComplete;
+
+  get #unsent() {
+    return this.#pending;
+  }
+
+  set #unsent(value) {
+    this.#pending = value;
+
+    if (this.#pending <= 0) {
+      this.#clearTimeout();
```

Comment:
Consider grouping lines 102-104 under a new private method like flushTransactions(), so that the purpose of these three lines is a bit more clear.

Also consider swapping lines 103 & 104 to help make this method less prone to abuse.